components-core/src/main/java/com/adyen/checkout/components/util/CountryUtils.kt

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright (c) 2020 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by josephj on 18/12/2020.
+ */
+
+package com.adyen.checkout.components.util
+
+import com.adyen.checkout.components.model.CountryInfo
+import java.util.*
+
+object CountryUtils {
+
+    /**
+     * Returns the list of supported countries to display.
+     *
+     * @param allowedISOCodes the list of ISO codes to filter the countries with. No filtering is done if this param is `null`.
+     * @return the list of countries.
+     */
+    @JvmStatic
+    fun getCountries(allowedISOCodes: List<String>? = null): List<CountryInfo> {
+        if (allowedISOCodes == null) return countries
+        return countries.filter { allowedISOCodes.contains(it.isoCode) }
+    }
+
+    /**
+     * Returns the localized display name of a country
+     *
+     * @param isoCode the ISO code of the country.
+     * @param locale the locale in which the country name should be localized.
+     * @return the localized country name.
+     */
+    @JvmStatic
+    fun getCountryName(isoCode: String, locale: Locale): String {
+        val countryLocale = Locale("", isoCode)
+        return countryLocale.getDisplayCountry(locale)
+    }
+
+    private val countries = listOf(
+        CountryInfo(isoCode = "AF", callingCode = "+93", emoji = "🇦🇫"),
+        CountryInfo(isoCode = "AL", callingCode = "+355", emoji = "🇦🇱"),
+        CountryInfo(isoCode = "DZ", callingCode = "+213", emoji = "🇩🇿"),
+        CountryInfo(isoCode = "AS", callingCode = "+1 684", emoji = "🇦🇸"),
+        CountryInfo(isoCode = "AD", callingCode = "+376", emoji = "🇦🇩"),
+        CountryInfo(isoCode = "AO", callingCode = "+244", emoji = "🇦🇴"),
+        CountryInfo(isoCode = "AI", callingCode = "+1 264", emoji = "🇦🇮"),
+        CountryInfo(isoCode = "AG", callingCode = "+1268", emoji = "🇦🇬"),
+        CountryInfo(isoCode = "AR", callingCode = "+54", emoji = "🇦🇷"),
+        CountryInfo(isoCode = "AM", callingCode = "+374", emoji = "🇦🇲"),
+        CountryInfo(isoCode = "AW", callingCode = "+297", emoji = "🇦🇼"),
+        CountryInfo(isoCode = "AU", callingCode = "+61", emoji = "🇦🇺"),
+        CountryInfo(isoCode = "AT", callingCode = "+43", emoji = "🇦🇹"),
+        CountryInfo(isoCode = "AZ", callingCode = "+994", emoji = "🇦🇿"),
+        CountryInfo(isoCode = "BS", callingCode = "+1 242", emoji = "🇧🇸"),
+        CountryInfo(isoCode = "BH", callingCode = "+973", emoji = "🇧🇭"),
+        CountryInfo(isoCode = "BD", callingCode = "+880", emoji = "🇧🇩"),
+        CountryInfo(isoCode = "BB", callingCode = "+1 246", emoji = "🇧🇧"),
```

Comment:
Some callingCodes have a space like this one `+1 242`, is that intentional?

Reply:
I just copied them from the API, maybe it's better to remove the spaces 👍